Brian/2_by_4_Decoder/decoder.h

```diff
@@ -0,0 +1,21 @@
+#include<systemc.h>
+
+SC_MODULE(decoder){
+//input and output ports
+sc_in<bool> d_e,d_s,d_r;
+sc_out<bool> d0,d1,d2,d3;
+//constructor: where the processes are bound to simulation kernel
+SC_CTOR(decoder){
+	SC_METHOD(decode);
+	sensitive<<d_e<<d_s<<d_r; //sensitive to i0 and i1 which is the input 
+	//dont_initialize();
+}
+
+~decoder(){
+//delete stuff :P
+}
+
+void decode(void){
+	
```

Comment:
Wrong file had been submitted but corrections have been made to it and push
to git .

On Thu, Feb 11, 2016 at 3:18 PM, David Muriuki <notifications@github.com>
wrote:

> In Brian/2_by_4_Decoder/decoder.h
> <https://github.com/Muriukidavid/cplusplus_examples/pull/26#discussion_r52593822>
> :
>
> > +//input and output ports
> > +sc_in<bool> d_e,d_s,d_r;
> > +sc_out<bool> d0,d1,d2,d3;
> > +//constructor: where the processes are bound to simulation kernel
> > +SC_CTOR(decoder){
> > +	SC_METHOD(decode);
> > +	sensitive<<d_e<<d_s<<d_r; //sensitive to i0 and i1 which is the input
> > +	//dont_initialize();
> > +}
> > +
> > +~decoder(){
> > +//delete stuff :P
> > +}
> > +
> > +void decode(void){
> > +	
>
> where is the implementation of this function??
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Muriukidavid/cplusplus_examples/pull/26/files#r52593822>
> .
>